src/main/java/com/minecolonies/coremod/client/gui/WindowHutBuilder.java

```diff
@@ -76,21 +137,67 @@ public int getElementCount()
             {
-                final Object obj = entries[index];
-
-                if (obj instanceof Map.Entry && ((Map.Entry) obj).getKey() instanceof String && ((Map.Entry) obj).getValue() instanceof Integer)
-                {
-                    @NotNull final String key = (String) ((Map.Entry) obj).getKey();
-                    final int value = (Integer) ((Map.Entry) obj).getValue();
-                    rowPane.findPaneOfTypeByID("resource", Label.class).setLabelText(key);
-                    rowPane.findPaneOfTypeByID("amount", Label.class).setLabelText(Integer.toString(value));
-                }
+                updateResourcePane(index, rowPane);
             }
         });
+
+        //Make sure we have a fresh view
+        MineColonies.getNetwork().sendToServer(new MarkBuildingDirtyMessage(this.building));
+    }
+
+    /**
+     * Update one row pad with its resource informations.
+     *
+     * @param index index in the list of resources.
+     * @param rowPane The Pane to use to display the information.
+     */
+    private void updateResourcePane(final int index, @NotNull final Pane rowPane)
+    {
+        final BuildingBuilder.BuildingBuilderResource resource = resources.get(index);
+        final Label resourceLabel = rowPane.findPaneOfTypeByID(RESOURCE_NAME, Label.class);
+        final Label neededLabel = rowPane.findPaneOfTypeByID(RESOURCE_AVAILABLE_NEEDED, Label.class);
+        final Button addButton = rowPane.findPaneOfTypeByID(RESOURCE_ADD, Button.class);
+
+        switch (resource.getAvailabilityStatus())
+        {
+            case DONT_HAVE:
+                addButton.disable();
+                resourceLabel.setColor(RED, RED);
+                neededLabel.setColor(RED, RED);
+                break;
+            case NEED_MORE:
+                addButton.enable();
+                resourceLabel.setColor(RED, RED);
+                neededLabel.setColor(RED, RED);
+                break;
+            case HAVE_ENOUGHT:
+                addButton.enable();
+                resourceLabel.setColor(DARKGREEN, DARKGREEN);
+                neededLabel.setColor(DARKGREEN, DARKGREEN);
+                break;
+            case NOT_NEEDED:
+            default:
+                addButton.disable();
+                resourceLabel.setColor(BLACK,BLACK);
+                neededLabel.setColor(BLACK,BLACK);
+                break;
+
+        }
+
+        //position the addRessource Button to the right
+        final int buttonX = rowPane.getWidth() - addButton.getWidth() - (rowPane.getHeight() - addButton.getHeight()) / 2;
+        final int buttonY = (rowPane.getHeight() - addButton.getHeight())/2;
+        addButton.setPosition(buttonX,buttonY);
+
+        resourceLabel.setLabelText(resource.getName());
+        final int missing = resource.getAvailable()-resource.getNeeded();
+        final String missingItems = (missing>0)?("+"+Integer.toString(missing)):Integer.toString(missing);
```

Comment:
can you leave some space to increase readability before and after "+" "-" "?" ">" ":"